deepdiff/serialization.py

```diff
@@ -310,12 +310,16 @@ def pretty(self):
             Item root[1] removed from set.
         """
         result = []
+        if prefix is None:
+            prefix = ''
         keys = sorted(self.tree.keys())  # sorting keys to guarantee constant order across python versions.
         for key in keys:
             for item_key in self.tree[key]:
                 result += [pretty_print_diff(item_key)]
 
-        return '\n'.join(result)
+        if callable(prefix):
+            return "\n".join(f"{prefix(diff=self)}{r}" for r in result)
+        return "\n".join(f"{prefix}{r}" for r in result)
```

Comment:
If you are concerned with the additional overhead of the f-string with no prefix being called every time, we can always move the `None` check down here and then restore the original `return` if it's `None`.

Reply:
If maintainer doesn't care, I don't care.